Error for NC_HDF5_FILE_INFO

Hello,
I'm getting an error that previously had not happened:
```
libtool: compile:  /opt/openmpi/bin/mpicc -fPIC -DHAVE_CONFIG_H -I. -I.. -I../include -L/o                                                                                                                                                   pt/openmpi/include -I/opt/hdf5/include -I/opt/openmpi/include -I/opt/hdf5/include -c hdf5f                                                                                                                                                   ile.c -MD  -fPIC -DPIC -o .libs/libnchdf5_la-hdf5file.o
"hdf5file.c", line 227: error: struct "NC_HDF5_FILE_INFO" has no field "http"
      ncurifree(hdf5_info->http.uri);
                           ^

"hdf5file.c", line 228: error: struct "NC_HDF5_FILE_INFO" has no field "http"
      NC_authfree(hdf5_info->http.auth);
                             ^

2 errors detected in the compilation of "hdf5file.c".
make[2]: *** [Makefile:578: libnchdf5_la-hdf5file.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
"nc4hdf.c", line 2291: warning: integer conversion resulted in a change of sign
      size_t maxobjs = -1;
                       ^

libtool: compile:  /opt/openmpi/bin/mpicc -fPIC -DHAVE_CONFIG_H -I. -I.. -I../include -L/o                                                                                                                                                   pt/openmpi/include -I/opt/hdf5/include -I/opt/openmpi/include -I/opt/hdf5/include -c hdf5a                                                                                                                                                   ttr.c -MD -o libnchdf5_la-hdf5attr.o >/dev/null 2>&1
libtool: compile:  /opt/openmpi/bin/mpicc -fPIC -DHAVE_CONFIG_H -I. -I.. -I../include -L/o                                                                                                                                                   pt/openmpi/include -I/opt/hdf5/include -I/opt/openmpi/include -I/opt/hdf5/include -c nc4in                                                                                                                                                   fo.c -MD -o libnchdf5_la-nc4info.o >/dev/null 2>&1
libtool: compile:  /opt/openmpi/bin/mpicc -fPIC -DHAVE_CONFIG_H -I. -I.. -I../include -L/o                                                                                                                                                   pt/openmpi/include -I/opt/hdf5/include -I/opt/openmpi/include -I/opt/hdf5/include -c nc4hd                                                                                                                                                   f.c -MD -o libnchdf5_la-nc4hdf.o >/dev/null 2>&1
make[2]: Leaving directory '/home/odyhpc/netcdf-c/libhdf5'
make[1]: *** [Makefile:722: all-recursive] Error 1
make[1]: Leaving directory '/home/odyhpc/netcdf-c'
make: *** [Makefile:567: all] Error 2
```
I noticed #1028 so one wouldn't probably expect this to be an issue. The system is running CentOS8 and netcdf is cloned from the Github page. Thanks.

--- a/libhdf5/hdf5file.c
+++ b/libhdf5/hdf5file.c
@@ -224,8 +224,10 @@ nc4_close_netcdf4_file(NC_FILE_INFO_T *h5, int abort, NC_memio *memio)
     NC4_clear_provenance(&h5->provenance);
 
     /* Free the http info */
+#ifdef ENABLE_HDF5_ROS3
     ncurifree(hdf5_info->http.uri);
     NC_authfree(hdf5_info->http.auth);
+#endif
 
     /* Close hdf file. It may not be open, since this function is also
      * called by NC_create() when a file opening is aborted. */

